python/ray/data/tests/test_tensor.py

```diff
@@ -40,7 +41,7 @@ def test_tensors_basic(ray_start_regular_shared):
     assert str(ds) == (
         "Dataset(num_rows=6, schema={data: numpy.ndarray(shape=(3, 5), dtype=int64)})"
     )
-    assert ds.size_bytes() == 5 * 3 * 6 * 8
+    assert math.isclose(ds.size_bytes(), 5 * 3 * 6 * 8, rel_tol=0.1)
```

Comment:
why do we need this tolerance now?

Reply:
Unlike `schema`, `count`, and `input_files`, I didn't update `size_bytes` to be lazy. So, it's now using the actual size which is slightly larger than the estimated size. 

I can update the code to make `size_bytes` lazy either in this PR or as a follow-up

Reply:
followup PR sounds good. let's also leave a comment here saying why we need the tolerance?